File: app/src/main/java/ru/binaryblitz/sportup/utils/LogUtil.java
package ru.binaryblitz.sportup.utils;

import android.util.Log;

@SuppressWarnings("unused")
public final class LogUtil {
    public static void logError(String logMessage) {
        Log.e(AppConfig.LogTag, logMessage);
    }

    public static void logError(byte[] logMessage) {
        String res = "";

        for (byte aLogMessage : logMessage) {
            res += Byte.toString(aLogMessage) + " , ";
        }

        Log.e(AppConfig.LogTag, res);
    }

    public static void logError(Integer logMessage) {
        Log.e(AppConfig.LogTag, "value: " + logMessage);
    }

    public static void logError(Double logMessage) {
        Log.e(AppConfig.LogTag, "value: " + logMessage);
    }

    public static void logInformation(String logMessage) {
        Log.i(AppConfig.LogTag, logMessage);
    }

    public static void logInformation(Integer logMessage) {
        Log.i(AppConfig.LogTag, "value: " + logMessage);
    }

    public static void logInformation(Double logMessage) {
        Log.i(AppConfig.LogTag, "value: " + logMessage);
    }

    public static void logException(Exception ex) {
        Log.e(AppConfig.LogTag, ex.getLocalizedMessage());
        ex.printStackTrace();
    }
}

File: app/src/main/java/ru/binaryblitz/SportUp/utils/LogUtil.kt
package ru.binaryblitz.SportUp.utils

import android.util.Log

object LogUtil {
    fun logError(logMessage: String) {
        Log.e(AppConfig.LogTag, logMessage)
    }

    fun logError(logMessage: ByteArray) {
        var res = ""

        for (aLogMessage in logMessage) {
            res += java.lang.Byte.toString(aLogMessage) + " , "
        }

        Log.e(AppConfig.LogTag, res)
    }

    fun logError(logMessage: Int?) {
        Log.e(AppConfig.LogTag, "value: " + logMessage!!)
    }

    fun logError(logMessage: Double?) {
        Log.e(AppConfig.LogTag, "value: " + logMessage!!)
    }

    fun logInformation(logMessage: String) {
        Log.i(AppConfig.LogTag, logMessage)
    }

    fun logInformation(logMessage: Int?) {
        Log.i(AppConfig.LogTag, "value: " + logMessage!!)
    }

    fun logInformation(logMessage: Double?) {
        Log.i(AppConfig.LogTag, "value: " + logMessage!!)
    }

    fun logException(ex: Exception) {
        Log.e(AppConfig.LogTag, ex.localizedMessage)
        ex.printStackTrace()
    }
}
